cilium-cli/encrypt/status.go

```diff
@@ -52,6 +64,49 @@ func (s *Encrypt) PrintEncryptStatus(ctx context.Context) error {
 	return printClusterStatus(cs, s.params.Output)
 }
 
+func (s *Encrypt) checkAndGetCiliumVersion(ctx context.Context, pods []corev1.Pod) (*semver.Version, error) {
+	// res contains data returned from cilium pod
+	type res struct {
+		version *semver.Version
+		err     error
+	}
+	resCh := make(chan res)
+	defer close(resCh)
+
+	// concurrently fetch state from each cilium pod
```

Comment:
I'm wondering if this isn't a bit over-engineered. I don't think we go to such lengths anywhere else in the CLI. What's the worse that can happen if we skip this?

Reply:
In the worst case, we can have a cluster with an upgrade in progress...
I agree that it's a pretty rare scenario.
Is it better to simplify the code and read the Cilium version from the first pod?

Reply:
So we'd perform a key rotation during an upgrade. Clearly uncovered in CI, but what would break? I guess if the upgrade is from v1.17 to v1.18, then they still can't be using the old key system (we removed support in v1.17). Is there something else in v1.17 we were worried about?

Reply:
I agree but then the question is: which version should we consider if there are pods with different versions?
I mean if we simply select the first pod it might still run 1.17 (which is not supported) but the second pod might be already updated to 1.18.

We can read all the versions and select the highest one, WDYT?

Reply:
I'm suggesting to check only one pod and if it's lower than v1.18, we bail out. Is there a scenario where that can cause issues?

Reply:
Not a big deal, user can always re-run the command.
Refactored.
I've also dropped the last commit as not needed with the current logic.